1. Given an integer, n, perform the following conditional actions: 

  If n is odd, print Weird

  If n is even and in the inclusive range of 2 to 5, print Not Weird

  If n is even and in the inclusive range of 6 to 20, print Weird

  If n is even and greater than 20, print Not Weird

In [ ]:
import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input().strip())
    if n % 2 == 1:
        print("Weird")
    elif n % 2 == 0 and n>=2 and n<=5:
        print("Not Weird")
    elif n % 2 == 0 and n>=6 and n<=20:
        print("Weird")
    elif n % 2 == 0 and n>20:
        print("Not Weird")

8
Weird


2. The provided code stub reads two integers from STDIN, a and b. 

  Add code to print three lines where:

  The first line contains the sum of the two numbers.

  The second line contains the difference of the two numbers (first - second).
  
  The third line contains the product of the two numbers.


In [ ]:
if __name__ == '__main__':
    a = int(input())
    b = int(input())
    print(a + b)
    print(a - b)
    print(a * b)

3
5
8
-2
15


3. The provided code stub reads two integers, a and b, from STDIN.

  Add logic to print two lines. The first line should contain the result of integer division,  a// b. The second line should contain the result of float division, a / b.

  No rounding or formatting is necessary.

In [ ]:
if __name__ == '__main__':
    a = int(input())
    b = int(input())
    print(a // b)
    print(a / b)


123
3
41
41.0


4. The provided code stub reads and integer, n, from STDIN. For all non-negative integers i < n, print i^2.

In [ ]:
if __name__ == '__main__':
    n = int(input())
    for i in range(0, n):
     print(i*i)
      
    
    

5
0
1
4
9
16


5. Given a year, determine whether it is a leap year. If it is a leap year, return the Boolean True, otherwise return False.

  Note that the code stub provided reads from STDIN and passes arguments to the is_leap function. It is only necessary to complete the is_leap function.

In [ ]:
def is_leap(year):
    leap = False
    if year % 4 == 0:
        leap = True
    if year % 100 == 0:
        if year % 400 == 0:
            leap = True
        else:
            leap = False  
    return leap

year = int(input())
print(is_leap(year))

1990
False


6. The included code stub will read an integer, n, from STDIN. Without using any string methods, try to print the following:

  123...n

  Note that "..." represents the consecutive values in between.

In [ ]:
if __name__ == '__main__':
    n = int(input())
    
    for i in range(1, n + 1):
      print(i, end='')

3
123

7. Let's learn about list comprehensions! You are given three integers x, y and z representing the dimensions of a cuboid along with an integer n. Print a list of all possible coordinates given by (i,j,k) on a 3D grid where the sum of i + j + k is not equal to n. Here, 0<=i<=;0<=j<=y;0<=k<=z. 
  Please use list comprehensions rather than multiple loops, as a learning exercise.

In [ ]:
if __name__ == '__main__':
    x = int(input())
    y = int(input())
    z = int(input())
    n = int(input())

    print ([[ i, j, k] for i in range( x + 1) for j in range( y + 1) for k in range ( z + 1 ) if ( ( i + j + k) != n )])

1
1
2
3
[[0, 0, 0], [0, 0, 1], [0, 0, 2], [0, 1, 0], [0, 1, 1], [1, 0, 0], [1, 0, 1], [1, 1, 0], [1, 1, 2]]


8. Given the participants' score sheet for your University Sports Day, you are required to find the runner-up score. You are given n scores. Store them in a list and find the score of the runner-up.

  Input Format

  The first line contains n. The second line contains an array A [] of n integers each separated by a space.

In [ ]:
if __name__ == '__main__':
    n = int(input())
    arr = map(int, input().split())

    print(sorted(set(arr), reverse=True)[1]) # sorted = ordena, set = quita duplicados, reverse = ordena de max a min


5
2 3 6 6 5
5


9. Given the names and grades for each student in a class of N students, store them in a nested list and print the name(s) of any student(s) having the second lowest grade.

  Note: If there are multiple students with the second lowest grade, order their names alphabetically and print each name on a new line.

  Example

  records = [["chi",20.0],["beta",50.0],["alpha",50.0]]

  The ordered list of scores is [20.0,50.0], so the second lowest score is 50.0. There are two students with that score: ["beta","alpha"]. Ordered alphabetically, the names are printed as:

  alpha

  beta

In [ ]:
if __name__ == '__main__':
    name_score = []
    for _ in range(int(input())):
        name = input()
        score = float(input())
        name_score.append([name,score])
    grade = sorted(list(set([x[1] for x in name_score])))
    sec_lowest = grade[1]
    student_list = []
    for student in name_score:
      if sec_lowest == student[1]:
        student_list.append(student[0])
    for name in sorted(student_list):
        print(name)     
  

      

5
Harry
37.21
Berry
37.21
Tina
37.2
Akriti
41
Harsh
39
Berry
Harry


10. The provided code stub will read in a dictionary containing key/value pairs of name:[marks] for a list of students. Print the average of the marks array for the student name provided, showing 2 places after the decimal.

In [32]:
if __name__ == '__main__':
    n = int(input())
    student_marks = {}
    for _ in range(n):
        name, *line = input().split()
        scores = list(map(float, line))
        student_marks[name] = scores
    query_name = input()

    if query_name in student_marks:
      average = round(sum(student_marks[query_name])/len(scores), 2)
print("%.2f" %average)


2
Harsh 25 26.5 28
Anurag 26 28 30
Harsh
26.50
